.github/workflows/ubuntu.yml

```diff
@@ -55,78 +55,107 @@ jobs:
       #
       - name: Build environment setup
         run: |
-          mkdir _build
+          mkdir build
           export TERM="vt100"
           echo "TERM=$TERM" >> $GITHUB_ENV
           echo "INSTALL_PATH=$(pwd)/_install" >> $GITHUB_ENV
 
       - name: Configure
         run: |
-          cd _build
+          cd build
           ../configure --enable-cobc-internal-checks \
                        --enable-hardening \
                        --prefix ${INSTALL_PATH}
           echo "VERSION=PACKAGE_VERSION" | cpp -P -imacros config.h | tr -d \" \
             >> $GITHUB_ENV
 
+      - name: Build
+        run: |
+          make -C build --jobs=$(($(nproc)+1))
+
       - name: Upload config.log
         uses: actions/upload-artifact@v4
         if: failure()
         with:
           name: config-${{ matrix.os }}-${{ github.job }}.log
-          path: _build/config.log
-
-      - name: Build
-        run: |
-          make -C _build --jobs=$(($(nproc)+1))
+          path: build/config.log
 
       # note: distcheck also creates the dist tarball
       - name: Build distribution archive & run tests
         run: |
-          make -C _build distcheck                                \
+          make -C build distcheck                                \
                          TESTSUITEFLAGS="--jobs=$(($(nproc)+1))"  \
                          --jobs=$(($(nproc)+1)) ||                \
-          make -C _build/gnucobol-$VERSION/_build/sub/tests check \
+          make -C build/gnucobol-$VERSION/build/sub/tests check \
                          TESTSUITEFLAGS="--recheck --verbose"
 
       - name: Upload testsuite.log
         uses: actions/upload-artifact@v4
-        if: failure()
+        # if: failure()   -> always upload as build result documentation
```

Comment:
I believe you have to explicitly specify `always()`

Reply:
nope, it is just step in the workflow as all the others, we only don't need to make that conditional

Reply:
My memory is vague about it, but I had the impression that if a step fails (and `continue-on-error` is unset or set to `false`), it skips all the subsequent steps ?

Reply:
Your memory is right, I've adjusted it accordingly.